hstream/src/HStream/Client/Types.hs

```diff
@@ -35,12 +35,14 @@ data HStreamSqlContext = HStreamSqlContext
 data HStreamSqlOpts = HStreamSqlOpts
   { _updateInterval :: Int
   , _retryTimeout   :: Int
+  , _execute        :: String
```

Comment:
Use `Maybe String` instead and throw an exception if statement is empty.

Reply:
if a statement is empty, it will invoke the interactive SQL cli. since the parser is default to empty string so the parser will not fail. does there means that it will be better to distinguish empty --execute from not provided the option?

Reply:
[f11cf40](https://github.com/hstreamdb/hstream/pull/1026/commits/f11cf401bf7ab3caa569824892d2bd905ea4d27b)

Reply:
[faee1e8](https://github.com/hstreamdb/hstream/pull/1026/commits/faee1e899f04ac35072a9b2bc8989a0609371bca)

Reply:
use `optional $ someparser`, https://hackage.haskell.org/package/base-4.17.0.0/docs/Control-Applicative.html#v:optional. The parser won't fail either, and this makes more sense.

Reply:
You can also checkout `Config.hs` (HStream.Server.Config). There are examples of `optional`。

Reply:
it has a check [here](https://github.com/hstreamdb/hstream/pull/1026/files#diff-9593fe69b5bcc3c830b5c1f9069da528eca4afaf30746d58f60708d05cd5e0feR118) which filter out all ASCII empty string and exit failure